ui/ducks/bridge/actions.ts

```diff
@@ -62,3 +166,359 @@ export const setToChain = (chainId: Hex) => {
+    }) => {
+      const hexChainId = new Numeric(
+        approval.chainId,
+        10,
+      ).toPrefixedHexString() as `0x${string}`;
+
+      // On Ethereum, we need to reset the allowance to 0 for USDT first if we need to set a new allowance
+      // https://www.google.com/url?q=https://docs.unizen.io/trade-api/before-you-get-started/token-allowance-management-for-non-updatable-allowance-tokens&sa=D&source=docs&ust=1727386175513609&usg=AOvVaw3Opm6BSJeu7qO0Ve5iLTOh
+      if (isEthUsdt(hexChainId, quoteResponse.quote.srcAsset.address)) {
+        await handleEthUsdtAllowanceReset({
+          approval,
+          hexChainId,
+          maxFeePerGas,
+          maxPriorityFeePerGas,
+        });
+      }
+
+      const txMeta = await handleTx({
+        txType: 'bridgeApproval',
+        txParams: approval,
+        maxFeePerGas,
+        maxPriorityFeePerGas,
+        meta: {
+          type: 'bridgeApproval', // TransactionType.bridgeApproval, // TODO
+          sourceTokenSymbol: quoteResponse.quote.srcAsset.symbol,
+        },
+      });
+
+      return txMeta.id;
+    };
+
+    const handleBridgeTx = async ({
+      approvalTxId,
+      maxFeePerGas,
+      maxPriorityFeePerGas,
+    }: {
+      approvalTxId: string | undefined;
+      maxFeePerGas: string | undefined;
+      maxPriorityFeePerGas: string | undefined;
+    }) => {
+      const txMeta = await handleTx({
+        txType: 'bridge',
+        txParams: quoteResponse.trade,
+        maxFeePerGas,
+        maxPriorityFeePerGas,
+        meta: {
+          // estimatedBaseFee: decEstimatedBaseFee,
+          // swapMetaData,
+          type: 'bridge', // TransactionType.bridge, // TODO add this type
+          sourceTokenSymbol: quoteResponse.quote.srcAsset.symbol,
+          destinationTokenSymbol: quoteResponse.quote.destAsset.symbol,
+          destinationTokenDecimals: quoteResponse.quote.destAsset.decimals,
+          destinationTokenAddress: quoteResponse.quote.destAsset.address,
+          approvalTxId,
+          // this is the decimal (non atomic) amount (not USD value) of source token to swap
+          swapTokenValue: new Numeric(quoteResponse.quote.srcTokenAmount, 10)
+            .shiftedBy(quoteResponse.quote.srcAsset.decimals)
+            .toString(),
+        },
+      });
+
+      return txMeta.id;
+    };
+
+    const addSourceToken = () => {
+      const sourceNetworkClientId: string = getSelectedNetworkClientId(state);
+      const {
+        address,
+        decimals,
+        symbol,
+        icon: image,
+        chainId,
+      } = quoteResponse.quote.srcAsset;
+
+      const srcHexChainId = new Numeric(
+        quoteResponse.quote.srcChainId,
+        10,
+      ).toPrefixedHexString() as `0x${string}`;
+      const tokenHexChainId = new Numeric(
+        chainId,
+        10,
+      ).toPrefixedHexString() as `0x${string}`;
+
+      if (tokenHexChainId !== srcHexChainId) {
+        throw new Error('Token chain ID does not match source chain ID');
```

Comment:
Does this show the exception screen to the user? Can we fail silently or log the error instead so the user gets routed to the Activity page anyway?

Reply:
Similar comment for the dest token

Reply:
No this does not show the exception screen to the user.

Reply:
Hmm I might have had this check initially for undefined/null reasons, but doesn't seem like it's relevant anymore, so I can just remove it.

Reply:
Removed in [3cdbb46](https://github.com/MetaMask/metamask-extension/pull/27262/commits/3cdbb46d152ae19165a73d5accbe761d1d679e25)